---
output: html_document
editor_options: 
  chunk_output_type: console
---

# Usage 

This workflow can be run using `Cromwell`:

The workflow can then be started with the following command:

```{bash, eval=FALSE}
java \
    -jar cromwell-<version>.jar \
    run \
    rnaseq.wdl \
    -o rnaseq.options.json  \
    -i rnaseq.json \

```

The `rnaseq.options.json` contains the following json:

```{bash, eval=FALSE}
{
	"final_workflow_outputs_dir": "results",
  	"use_relative_output_paths": true,
    "default_runtime_attributes": {
        "docker": "dcassol/wdl_rnaseq"
    }
}
```

The `rnaseq.options.json` will make sure all outputs will be copy to 
`./results`. 

## Inputs

Inputs are provided through a `JSON` file. 

Please note here that for each major step, we are providing subdirectories for the 
output file. Also, the sample data file is provided here:

```{bash, eval=FALSE}
{
  "rnaseq.outdir_fastqc": "fastqc/",
  "rnaseq.outdir_trimmomatic": "trimmomatic/",
  "rnaseq.outdir_hs2b": "data_reference/", 
  "rnaseq.outdir_hisat2": "hisat2/",
  "rnaseq.inputSamplesFile": "./samples.txt",
}
```

Trimmomatic inputs:

```{bash, eval=FALSE}
{
  "rnaseq.trimmomatic_path": "trimmomatic",
  "rnaseq.seqType": "PE",
  "rnaseq.TRIM_threads": 2,
  "rnaseq.Phred": "-phred64",
  "rnaseq.truseq_pe_adapter": "param/adapters/TruSeq3-PE.fa",
  "rnaseq.leading": 3,
  "rnaseq.trailing": 3,
  "rnaseq.window": 4,
  "rnaseq.quality": 5,
  "rnaseq.minLength": 35,
}
```

`Hisat2` inputs: 

```{bash, eval=FALSE}
{
  "rnaseq.reference": "data/tair10.fasta",
  "rnaseq.sortThreads": 2,
  "rnaseq.hisat2threads": 2,
}
```

Quantification inputs:

```{bash, eval=FALSE}
{
  "rnaseq.gff": "data/tair10.gff",
  "rnaseq.dataSource": "TAIR",
  "rnaseq.db_path": "tair10.sqlite",
  "rnaseq.script_quant": "tasks/scripts/quantification.doc.R",
  "rnaseq.targets": "targetsPE.txt",
  "rnaseq.script_deg": "tasks/scripts/deg.doc.R"
}
```

## Outputs

This workflow will produce a number of directories and files:

- `results/data_reference`
    - `Hisat2` index files.
    
- `results/fastqc`
    - Contains a folder per sample with QC metrics.
    
- `results/trimmomatic`
    - Contains all the four output files for each sample:
        - *_R1_paired.fq.gz - for paired forward reads
        - *_R1_unpaired.fq.gz - for unpaired forward reads
        - *_R2_paired.fq.gz - for paired reverse reads
        - *_R2_unpaired.fq.gz - for unpaired reverse reads

- `results/hisat2`
    - Contains a BAM, index, and summary report for each sample.

- `results/countDFeByg.xls`: contain the count reads values for all samples.
- `results/rpkmDFeByg.xls`: contain the RPKM values for all samples.

